clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -104,6 +104,44 @@ public void unschedule(DelayedTask task) {
         return future;
     }
 
+    /**
+     * Similar to the send() method, except this method retries send() on SendFailedException
+     * several times
+     * @param node The destination of the request
+     * @param api The Kafka API call
+     * @param request The request payload
+     * @return A future which indicates the result of the send
+     */
+    public RequestFuture<ClientResponse> sendWithRetries(final Node node,
+                                                         final ApiKeys api,
+                                                         final AbstractRequest request) {
+        final RequestFuture<ClientResponse> retFuture = new RequestFutureCompletionHandler();
+        RequestFuture<ClientResponse> future = this.send(node, api, request);
+
+        future.addListener(new RequestFutureListener<ClientResponse>() {
+            final int maxRetries = 10;
```

Comment:
Minor: perhaps support a variant which allows this value to be passed as an argument? Might also be a good idea to store the value 10 as a constant static field (e.g. DEFAULT_RETRIES).